src/WorkflowManager/Controllers/WorkflowsController.cs

```diff
@@ -107,4 +107,46 @@ public async Task<IActionResult> CreateAsync([FromBody] Workflow workflow)
             return Problem($"Unexpected error occured: {e.Message}", $"/workflows", 500);
         }
     }
+
+    /// <summary>
+    /// Updates a workflow and creates a new revision
+    /// </summary>
+    /// <param name="workflow">The Workflow.</param>
+    /// <returns>The ID of the created Workflow.</returns>
+    [Route("{id}")]
+    [HttpPut]
```

Comment:
I think these 2 lines can be combined into 1 as: `[HttpPut("{id}")]`

Also, do we want to decorate all APIs with: `Consumes`, `Produces`, `ProducesResponseType` attributes?